eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -46,6 +47,32 @@ case class RemoteCommit(index: Long, spec: CommitmentSpec, txid: ByteVector32, r
 case class WaitingForRevocation(nextRemoteCommit: RemoteCommit, sent: CommitSig, sentAfterLocalCommitIndex: Long, reSignAsap: Boolean = false)
 // @formatter:on
 
+trait ChannelCommitments {
+  def getOutgoingHtlcCrossSigned(htlcId: Long): Option[UpdateAddHtlc]
+
+  def getIncomingHtlcCrossSigned(htlcId: Long): Option[UpdateAddHtlc]
+
+  def timedOutOutgoingHtlcs(blockheight: Long): Set[UpdateAddHtlc]
+
+  def localNodeId: PublicKey
+
+  def remoteNodeId: PublicKey
+
+  def capacity: MilliSatoshi
+
+  def fundingTxId: ByteVector32
+
+  def availableBalanceForReceive: MilliSatoshi
+
+  def availableBalanceForSend: MilliSatoshi
+
+  def originChannels: Map[Long, Origin]
+
+  def channelId: ByteVector32
+
+  def announceChannel: Boolean
+}
```

Comment:
Is this really all you need for hosted channels?
To help my understanding, are hosted channels using a different channel FSM than the one in eclair-core (I'd guess yes)?